Link failure building for Fedora

### Description

Trying to update the Fedora package to 3.5.0 we're getting:
```
 77%] Linking CXX executable ../../../../../bin/bsgm_app
cd /home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/contrib/brl/bseg/sdet/tests && /usr/bin/g++ -DCL_TARGET_OPENCL_VERSION=300 -DHAS_OPENCL=1 -DVXL_WARN_DEPRECATED -DVXL_WARN_DEPRECATED_ONCE -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/vcl -I/home/orion/fedora/vxl/vxl-3.5.0/vcl -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/core -I/home/orion/fedora/vxl/vxl-3.5.0/core -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/contrib/brl/bbas -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bseg -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bpro -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/mul -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/gel -I/home/orion/fedora/vxl/vxl-3.5.0/v3p -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bseg/sdet -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bseg/brip -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/brad -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bres -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/contrib/brl/bbas/bres -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bsta -I/home/orion/fedora/vxl/vxl-3.5.0/core/vnl/algo -I/home/orion/fedora/vxl/vxl-3.5.0/v3p/netlib -I/home/orion/fedora/vxl/vxl-3.5.0/core/vnl -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/core/vnl -I/home/orion/fedora/vxl/vxl-3.5.0/core/vbl -I/home/orion/fedora/vxl/vxl-3.5.0/core/vnl/io -I/home/orion/fedora/vxl/vxl-3.5.0/core/vsl -I/home/orion/fedora/vxl/vxl-3.5.0/core/vgl -I/home/orion/fedora/vxl/vxl-3.5.0/v3p/clipper -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bsta/algo -I/home/orion/fedora/vxl/vxl-3.5.0/core/vgl/algo -I/home/orion/fedora/vxl/vxl-3.5.0/core/vgl/io -I/home/orion/fedora/vxl/vxl-3.5.0/core/vil -I/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/core/vil -I/home/orion/fedora/vxl/vxl-3.5.0/v3p/openjpeg2 -I/home/orion/fedora/vxl/vxl-3.5.0/core/vpgl -I/home/orion/fedora/vxl/vxl-3.5.0/core/vul -I/home/orion/fedora/vxl/vxl-3.5.0/core/vpgl/algo -I/home/orion/fedora/vxl/vxl-3.5.0/core/vpgl/file_formats -I/home/orion/fedora/vxl/vxl-3.5.0/core/vbl/io -I/home/orion/fedora/vxl/vxl-3.5.0/core/vpl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/gel/gevd -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/gel/vtol -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/gel/vdgl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/gel/vsol -I/home/orion/fedora/vxl/vxl-3.5.0/core/vil1 -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bsol -I/home/orion/fedora/vxl/vxl-3.5.0/core/vil/algo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bil/algo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/btol -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bdgl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bvgl/algo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bvgl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bnabo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bvrml -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bkml -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bnl -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/bnl/algo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/imesh/algo -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/imesh -I/home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bbas/brdb -I/home/orion/fedora/vxl/vxl-3.5.0/core/testlib -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wno-complain-wrong-lang -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fpermissive  -O2 -g -DNDEBUG -std=c++11 -MD -MT contrib/brl/bseg/sdet/tests/CMakeFiles/sdet_test_all.dir/test_region_classifier.cxx.o -MF CMakeFiles/sdet_test_all.dir/test_region_classifier.cxx.o.d -o CMakeFiles/sdet_test_all.dir/test_region_classifier.cxx.o -c /home/orion/fedora/vxl/vxl-3.5.0/contrib/brl/bseg/sdet/tests/test_region_classifier.cxx
cd /home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/contrib/brl/bseg/bsgm/app && /usr/bin/cmake -E cmake_link_script CMakeFiles/bsgm_app.dir/link.txt --verbose=1
/usr/bin/g++ -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wno-complain-wrong-lang -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fpermissive  -O2 -g -DNDEBUG -Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes  CMakeFiles/bsgm_app.dir/bsgm_app.cxx.o -o ../../../../../bin/bsgm_app  -Wl,-rpath,/home/orion/fedora/vxl/vxl-3.5.0/redhat-linux-build/lib ../../../../../lib/libbsgm.so.3.5.0.0 ../../../../../lib/libvidl_pro.so.3.5.0.0 ../../../../../lib/libbprb.so.3.5.0.0 ../../../../../lib/libbxml.so.3.5.0.0 /usr/lib64/libexpatpp.so ../../../../../lib/libbrdb.so.3.5.0.0 ../../../../../lib/libvidl.so.3.5.0.0 /usr/lib64/libdc1394.so ../../../../../lib/libvil_io.so.3.5.0.0 ../../../../../lib/libbpgl_algo.so.3.5.0.0 ../../../../../lib/libbpgl.so.3.5.0.0 ../../../../../lib/libvpgl_io.so.3.5.0.0 ../../../../../lib/libbvgl.so.3.5.0.0 ../../../../../lib/libbnabo.so.3.5.0.0 ../../../../../lib/libbvrml.so.3.5.0.0 -lexpatpp ../../../../../lib/librrel.so.3.5.0.0 ../../../../../lib/libbrip.so.3.5.0.0 ../../../../../lib/libbrad.so.3.5.0.0 ../../../../../lib/libvpgl_algo.so.3.5.0.0 ../../../../../lib/libvpgl_file_formats.so.3.5.0.0 ../../../../../lib/libbres.so.3.5.0.0 ../../../../../lib/libbsta_algo.so.3.5.0.0 ../../../../../lib/libvpl.so.3.5.0.0 ../../../../../lib/libgevd.so.3.5.0.0 ../../../../../lib/libvtol.so.3.5.0.0 ../../../../../lib/libvdgl.so.3.5.0.0 ../../../../../lib/libbsol.so.3.5.0.0 ../../../../../lib/libvil1.so.3.5.0.0 ../../../../../lib/libbil_algo.so.3.5.0.0 ../../../../../lib/libvsol.so.3.5.0.0 ../../../../../lib/libvbl_io.so.3.5.0.0 ../../../../../lib/libbsta.so.3.5.0.0 ../../../../../lib/libvnl_io.so.3.5.0.0 ../../../../../lib/libbjson.so.3.5.0.0 ../../../../../lib/libvpgl.so.3.5.0.0 ../../../../../lib/libvul.so.3.5.0.0 ../../../../../lib/libvbl.so.3.5.0.0 ../../../../../lib/libvgl_io.so.3.5.0.0 ../../../../../lib/libvsl.so.3.5.0.0 ../../../../../lib/libvgl_algo.so.3.5.0.0 ../../../../../lib/libvil_algo.so.3.5.0.0 ../../../../../lib/libvil.so.3.5.0.0 /usr/lib64/libjpeg.so /usr/lib64/libtiff.so /usr/lib64/libgeotiff.so /usr/lib64/libpng.so /usr/lib64/libz.so ../../../../../lib/libopenjpeg2.so.2.0.0 ../../../../../lib/libvnl_algo.so.3.5.0.0 ../../../../../lib/libv3p_netlib.so.3.5.0.0 ../../../../../lib/libvnl.so.3.5.0.0 ../../../../../lib/libvgl.so.3.5.0.0 ../../../../../lib/libvcl.so.3.5.0.0 -lm ../../../../../lib/libclipper.so.3.5.0.0
/usr/bin/ld: ../../../../../lib/libbsgm.so.3.5.0.0: undefined reference to `acal_f_utils::read_affine_cameras(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::map<unsigned long, vpgl_affine_camera<double>, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, vpgl_affine_camera<double> > > >&)'
/usr/bin/ld: ../../../../../lib/libbsgm.so.3.5.0.0: undefined reference to `acal_f_utils::datetime_from_iname(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, dtime&)'
/usr/bin/ld: ../../../../../lib/libbsgm.so.3.5.0.0: undefined reference to `dtime::time_diff(dtime const&, dtime const&)'
collect2: error: ld returned 1 exit status
```
### Steps to Reproduce

```
+ CFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3
-Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x8
6-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-point
er '
+ export CFLAGS
+ CXXFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=
3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=
x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-poi
nter '
+ export CXXFLAGS
+ FFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTION
S -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fas
ynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -I/usr/lib64/gfortran/modules '
+ export FFLAGS
+ FCFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -I/usr/lib64/gfortran/modules '
+ export FCFLAGS
+ VALAFLAGS=-g
+ export VALAFLAGS
+ RUSTFLAGS='-Copt-level=3 -Cdebuginfo=2 -Ccodegen-units=1 -Cstrip=none -Cforce-frame-pointers=yes -Clink-arg=-specs=/usr/lib/rpm/redhat/redhat-package-notes --cap-lints=warn'
+ export RUSTFLAGS
+ LDFLAGS='-Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes '
+ export LDFLAGS
+ LT_SYS_LIBRARY_PATH=/usr/lib64:
+ export LT_SYS_LIBRARY_PATH
+ CC=gcc
+ export CC
+ CXX=g++
+ export CXX
+ /usr/bin/cmake -S . -B redhat-linux-build -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -DVXL_INSTALL_LIBRARY_DIR:PATH=lib64 -DBUILD_SHARED_LIBS:BOOL=ON -DVXL_USE_GEOTIFF:BOOL=ON -DVXL_FORCE_B3P_EXPAT:BOOL=OFF -DVXL_FORCE_V3P_DCMTK:BOOL=ON -DVXL_FORCE_V3P_GEOTIFF:BOOL=OFF -DVXL_FORCE_V3P_JPEG:BOOL=OFF -DVXL_FORCE_V3P_PNG:BOOL=OFF -DVXL_FORCE_V3P_TIFF:BOOL=OFF -DVXL_FORCE_V3P_ZLIB:BOOL=OFF -DVXL_FORCE_V3P_RPLY:BOOL=OFF -DVXL_FORCE_V3P_OPENJPEG2:BOOL=ON -DVXL_USING_NATIVE_ZLIB:BOOL=ON -DVXL_USING_NATIVE_JPEG:BOOL=ON -DVXL_USING_NATIVE_PNG:BOOL=ON -DVXL_USING_NATIVE_TIFF:BOOL=ON -DVXL_USING_NATIVE_GEOTIFF:BOOL=ON -DVXL_USING_NATIVE_EXPAT:BOOL=ON -DVXL_USING_NATIVE_EXPATPP:BOOL=ON -DVXL_USING_NATIVE_SHAPELIB:BOOL=ON -DVXL_USING_NATIVE_BZLIB2:BOOL=ON -DVXL_BUILD_VGUI:BOOL=OFF -DVXL_BUILD_BGUI3D:BOOL=OFF -DVXL_BUILD_OXL:BOOL=ON -DVXL_BUILD_BRL:BOOL=ON -DVXL_BUILD_BRL_PYTHON:BOOL=OFF -DVXL_BUILD_GEL:BOOL=ON -DVXL_BUILD_PRIP:BOOL=ON -DVXL_BUILD_CONVERSIONS:BOOL=ON -DVXL_BUILD_CUL:BOOL=ON -DVXL_BUILD_RPL:BOOL=ON -DVXL_BUILD_CONTRIB:BOOL=ON -DVXL_BUILD_CORE_SERIALISATION:BOOL=ON -DVXL_BUILD_CORE_GEOMETRY:BOOL=ON -DVXL_BUILD_CORE_IMAGING:BOOL=ON -DVXL_BUILD_CORE_NUMERICS:BOOL=ON -DVXL_BUILD_CORE_PROBABILITY:BOOL=ON -DVXL_BUILD_CORE_UTILITIES:BOOL=ON -DVXL_BUILD_CORE_VIDEO:BOOL=ON -DVXL_BUILD_EXAMPLES:BOOL=OFF -DBUILD_TESTING:BOOL=ON -DVXL_BUILD_DOCUMENTATION:BOOL=ON -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo '-DCMAKE_CXX_FLAGS:STRING=-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wno-complain-wrong-lang -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fpermissive' -DVNL_CONFIG_LEGACY_METHODS:BOOL=ON
```


### Versions

3.5.0

### Environment

Fedora Rawhide

--- a/contrib/brl/bseg/bsgm/CMakeLists.txt
+++ b/contrib/brl/bseg/bsgm/CMakeLists.txt
@@ -16,7 +16,7 @@ aux_source_directory(Templates bsgm_sources)
 
 vxl_add_library(LIBRARY_NAME bsgm LIBRARY_SOURCES ${bsgm_sources})
 
-target_link_libraries( bsgm vidl_pro ${VXL_LIB_PREFIX}bpgl_algo ${VXL_LIB_PREFIX}brip ${VXL_LIB_PREFIX}bsta ${VXL_LIB_PREFIX}bjson ${VXL_LIB_PREFIX}vpgl ${VXL_LIB_PREFIX}vgl_io ${VXL_LIB_PREFIX}vgl_algo ${VXL_LIB_PREFIX}vil_algo ${VXL_LIB_PREFIX}vnl_algo ${VXL_LIB_PREFIX}vnl ${VXL_LIB_PREFIX}vgl ${VXL_LIB_PREFIX}vil ${VXL_LIB_PREFIX}vul ${VXL_LIB_PREFIX}vcl)
+target_link_libraries( bsgm vidl_pro ${VXL_LIB_PREFIX}acal ${VXL_LIB_PREFIX}bpgl_algo ${VXL_LIB_PREFIX}brip ${VXL_LIB_PREFIX}bsta ${VXL_LIB_PREFIX}bjson ${VXL_LIB_PREFIX}vpgl ${VXL_LIB_PREFIX}vgl_io ${VXL_LIB_PREFIX}vgl_algo ${VXL_LIB_PREFIX}vil_algo ${VXL_LIB_PREFIX}vnl_algo ${VXL_LIB_PREFIX}vnl ${VXL_LIB_PREFIX}vgl ${VXL_LIB_PREFIX}vil ${VXL_LIB_PREFIX}vul ${VXL_LIB_PREFIX}vcl)
 
 add_subdirectory( app )
 

